authorize.py

```diff
@@ -0,0 +1,75 @@
+"""Authenticate the user with the Sam service."""
+
+from collections.abc import Callable
+from flask import request
+from functools import wraps
+from requests import get, Response
+from requests.exceptions import ConnectionError
+import werkzeug.exceptions as e
+
+
+def authorized(sam_root: str) -> Callable[..., any]:
+    """Authorization decorator.
+
+    Decorated routes will precheck the request to see if the caller is authorized to perform the operation.
+    Taken from example in Sanic documentation: `https://sanic.readthedocs.io/en/latest/sanic/decorators.html`
+    """
+    def decorator(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            # Check the Terra authorization service SAM for user auth status
+            user_is_authorized = _check_sam_authorization(sam_root)
+            if user_is_authorized:
+                # run the handler method and return the response
+                response = f(*args, **kwargs)
+                return response
+            else:
+                raise e.Forbidden
+        return decorated_function
+    return decorator
+
+
+def _check_sam_authorization(sam_root: str) -> bool:
+    """Query Terra's authorization service SAM to determine user authorization status. Return auth status boolean."""
+    sam_url = sam_root + '/register/user/v2/self/info'
+
+    # Well-formed requests must contain an authorization header
+    if 'authorization' not in request.headers:
+        raise e.Unauthorized('Request requires authorization header supplying Oauth2 bearer token')
+    try:
+        sam_response = get(sam_url, headers={'authorization': request.headers['authorization']})
+        return _process_sam_response(sam_response)
+    except ConnectionError:
+        raise e.ServiceUnavailable('Service Unavailable. Unable to contact authorization service')
+
+
+def _process_sam_response(sam_response: Response) -> bool:
```

Comment:
Same comment here, the return type is the union of a boolean, and errors

Reply:
I think that `raise` is not a return type. Reading thru instructions online, it seems like the thing to do is to add `:raises` clauses to the docstrings?

Reply:
Done! All docstrings updated per Google style